packages/viz/src/BarChart/stories/BarChart.stories.tsx

```diff
@@ -494,3 +497,110 @@ export const CustomEchartsOptions: StoryObj<HvBarChartProps> = {
   },
   render: () => <CustomEchartsOptionsStory />,
 };
+
+export const WithFiltering: StoryObj<HvBarChartProps> = {
+  parameters: {
+    docs: {
+      description: {
+        story:
+          "This sample illustrates how to filter the data using the `filters` prop.",
+      },
+    },
+  },
+  render: () => {
+    const [sales, setSales] = useState([0, 7000]);
+    const [country, setCountry] = useState("");
+    const [filters, setFilters] = useState<HvChartFilter[]>([]);
+
+    const data = [
+      {
+        label: "Portugal",
+        value: 3600,
+      },
+      {
+        label: "USA",
+        value: 1500,
+      },
+      {
+        label: "India",
+        value: 6700,
+      },
+      {
+        label: "China",
+        value: 4500,
+      },
+      {
+        label: "France",
+        value: 3200,
+      },
+      {
+        label: "UK",
+        value: 2700,
+      },
+      {
+        label: "Japan",
+        value: 5800,
+      },
+    ];
+
+    useEffect(() => {
```

Comment:
I think we can remove this `useEffect` and only have one state, `filters`. Then, inside `handleSliderChange` and `handleCountryChange` we could do:

```
// handleCountryChange
setFilters((prevState) => ([
        ...prevState,
         {
           field: "Country",
           operation: "contains",
           value: country,
         }
       ]))

// handleSliderChange
setFilters((prevState) => ([
        ...prevState,
         {
           field: "Sales",
           operation: "between",
           value: [sales[0], sales[1]],
         },
       ]))
```

Less re-renders and states to manage.

Reply:
This is [derived state](https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state) - we don't need state

```tsx
const filters = [
        {
          field: "Country",
          operation: "contains",
          value: country,
        },
        {
          field: "Sales",
          operation: "between",
          value: [sales[0], sales[1]],
        },
      ];
```